File: Assignment-SEM05-2019/PLP/gst-c/gst_item.c
#include "gst_item.h"

#include <assert.h>
#include <stdio.h>
#include <stdlib.h>

float get_slab_tax(int slab_id) {
    return slab_tax[slab_id];
}

double get_mrp(gst_item* item) {
    assert(item != NULL);

    return item->price + get_slab_tax(item->gst_slab_id) * item->price;
}

void* create_gst_item(int item_id, int slab_id, double price) {
    gst_item* item;
    if ((item = malloc(sizeof *item)) != NULL) {
        item->item_id = item_id;
        item->gst_slab_id = slab_id;
        item->price = price;
        return item;
    }

    return NULL;
}

void free_gst_item(void* item) {
    gst_item* myitem = item;

    free(myitem);
}

File: Assignment-SEM05-2019/PLP/GSTJava/src/app/GSTItem.java
package app;

/**
 * GSTItem
 */
public class GSTItem {
    private int item_id;
    private int gst_slab_id;
    private double price;

    GSTItem(int item_id, int gst_slab_id, double price) {
        this.item_id = item_id;
        this.gst_slab_id = gst_slab_id;
        this.price = price;
    }

    public static float slab_tax[] = { 0.0f, 0.05f, 0.12f, 0.18f, 0.28f };

    public double get_mrp() {
        return this.price + slab_tax[this.gst_slab_id] * this.price;
    }
}